tests/framework/utils/mysql_helper.go

```diff
@@ -95,17 +95,17 @@ func (h *MySQLHelper) InsertRandomData() sql.Result {
 func (h *MySQLHelper) SelectRandomData(limit int) *sql.Rows {
 	query := fmt.Sprintf("SELECT * FROM  LONGHAUL ORDER BY RAND() LIMIT %d", limit)
 	rows, err := h.DB.Query(query)
+	defer rows.Close()
```

Comment:
will `rows` be nil when `err != nil`? If so, the `defer` might need to go after the err check

Reply:
yes, good catch, moving defer after err check